src/Rings/mpoly-graded.jl

```diff
@@ -1612,6 +1612,349 @@ function Base.show(io::IO, h::HilbertData)
   print(io, "Hilbert Series for $(h.I), data: $(h.data), weights: $(h.weights)")  ###new
 end
 
+########################################################################
+# Compute the Hilbert series from the monomial leading ideal using 
+# different bisection strategies along the lines of 
+# [Kreuzer, Robbiano: Computational Commutative Algebra 2, Springer]
+# Section 5.3.
+########################################################################
+exp_vec(f::MPolyElem) = first(exponent_vectors(f))
+
+_divides(a::Vector{Int}, b::Vector{Int}) = all(x->(x>=0), a-b)
+
+function _gcd(a::Vector{Vector{Int}}) 
```

Comment:
You only ever use this for vectors `a` of length two, so perhaps write a function that takes to `Vector{Int}` instead? that could then potentially be quite a bit quicker.